code/game/machinery/doors/spacepod.dm

```diff
@@ -15,7 +15,9 @@
 	return 0
 
 /obj/structure/spacepoddoor/Destroy()
-	air_update_turf(1)
+	var/turf/T = get_turf(src)
+	spawn(0)
```

Comment:
I won't allow this. This is gross.

If the problem is an order of operations things, then try
```
 /obj/structure/spacepoddoor/Destroy()
	. = ..()
	air_update_turf(TRUE)
```

if that fails, cache the turf locally before `. = ..()` then call the update turf process later.

Reply:
won't work due to the structure not being deleted in the same proc. It will get the flag to be deleted instead

Reply:
Uh, no, this isn't correct.

That's not how `air_update_turf` works.

This is what happens:

```
/turf/proc/air_update_turf(var/command = 0)
	if(command)
		CalculateAdjacentTurfs()
	if(SSair)
		SSair.add_to_active(src, command)
```
```
/turf/proc/CalculateAdjacentTurfs()
	atmos_adjacent_turfs_amount = 0
	for(var/direction in cardinal)
		var/turf/T = get_step(src, direction)
		if(!istype(T))
			continue
		var/counterdir = get_dir(T, src)
		if(CanAtmosPass(T))
			atmos_adjacent_turfs_amount += 1
			atmos_adjacent_turfs |= direction
			if(!(T.atmos_adjacent_turfs & counterdir))
				T.atmos_adjacent_turfs_amount += 1
			T.atmos_adjacent_turfs |= counterdir
		else
			atmos_adjacent_turfs &= ~direction
			if(T.atmos_adjacent_turfs & counterdir)
				T.atmos_adjacent_turfs_amount -= 1
			T.atmos_adjacent_turfs &= ~counterdir
```

`CanAtmosPass` sometimes checks for density, but usually it does not.

The reason this bug occurs is that when the object is destroyed, it calls `air_upate_turf` or `T.air_update_turf` on the object immediately, but the object is still there in most cases, so nothing changes; then `return ..()` is called.

the point of utilizing `. = ..()` is that it runs the parent `Destroy`'s first, while returning their value, then runs what's below it.

By doing something like:

```
/obj/structure/holosign/barrier/atmos/Destroy()
	var/turf/T = get_turf(src)
	. = ..()
	T.air_update_turf(TRUE)
	return ..()
```

the item doesn't exist on the turf (only in null space, briefly), so `air_update_turf` occurs properly.

That said, you HAVE to cache the turf variable locally prior to it being set to null `loc`, or else you'll be running `air_update_turf` on something that doesn't exist, so it'll runtime.

Reply:
Ill test it once more. I tested this at first and found it not working in my case. I might have missed something. The caching of the turf I did but might be something else.

Reply:
I tested it out with the atmos barrier and it resolved the issue just fine for it.

Reply:
Yep you're right. I must have fucked up somewhere while implementing it like that. Will change the others now